---
title: "GSOA Report"
output: 
  flexdashboard::flex_dashboard:
    theme: cerulean
    vertical_layout: scroll
    orientation: rows
    source_code: embed
params: 
    data1: NA
    data2: NA
    data3: NA
    data_files: NA
    data4: NA
    class: NA
    hallmarks: NA
    results_hallmark: NA
    alg: NA
    var: NA
    lowexpress: NA
    crossval: NA
    iterations: NA
    GSOA: NA
    genesets: NA

---

```{r include = FALSE}
suppressMessages(library(googleVis))
library(flexdashboard)
op <- options(gvis.plot.tag="chart")
```


# Your Analysis {data-icon="fa-user"}

Row {data-width=300}
-------------------------------------

### Total Number of Gene Sets GSOA Ran
```{r}

#read in GSOA results

#your_GSOA_results= read.table("GSOA_P53_C2.txt" ,header=T,sep='\t', quote="", row.names = 1)
your_GSOA_results= read.table(params$data1 ,header=T,sep='\t', quote="", row.names = 1)

total = nrow(your_GSOA_results)
gauge(total, min = 0, max = nrow(your_GSOA_results) , label = "Gene Sets" )

```
    
### Number of Significant Gene Sets (P-Value)
    
```{r}
sig_pval = nrow(your_GSOA_results[ which(your_GSOA_results$p.value <= 0.05),])

gauge(sig_pval, min = 0, max = nrow(your_GSOA_results) , label = "P-value < 0.05", gaugeSectors(
  success = c(0, total/3), warning = c(total/3+1, total/2), danger = c(total/2+1, sig_pval)))

```

### Number of Significant Gene Sets (FDR)

```{r}
sig_fdr = nrow(your_GSOA_results[ which(your_GSOA_results$FDR <= 0.05),])
gauge(sig_fdr, min = 0, max = 100 , label = "FDR < 0.05", gaugeSectors(
  success = c(0, total/3), warning = c(total/3+1, total/2), danger = c(total/2+1, sig_pval)))
```

Row {data-height=575}
-------------------------------------

### Top 20 Ranked Hallmark Gene Sets

```{r, results='asis', echo=FALSE}

your_GSOA_results_bar= your_GSOA_results
# hack
your_GSOA_results_bar[,1]=rownames(your_GSOA_results)

# plot the bar plot
#Bar1 <- gvisBarChart(your_GSOA_results_bar[1:20,], xvar= "AUC", yvar=c("Rank", "p.value"), options= list(fontSize=10.5, width = 1400, height = 550, title= "Top 20 Gene Sets", legend="right",hAxes="[{title:'Rank', titleTextStyle: {color: 'black', fontSize:14}}]", titleTextStyle="{color:'Black',fontName:'Arial',fontSize:18}",isStacked =  FALSE, chartArea= "{ width: '35%', height: '75%'}" ) )
#plot(Bar1)

Bar1 <- gvisBarChart(your_GSOA_results_bar[1:20,], "AUC", c("Rank", "p.value"),options=list(title= "Top 20 Gene Sets",
                         series="[{targetAxisIndex: 1},{targetAxisIndex:0}]",hAxis="{title:'P-value',   titleTextStyle:{color:'blue', fontSize:14}}" , titleTextStyle="{color:'Black',fontName:'Arial',fontSize:18}", chartArea= "{ width: '45%', height: '75%'}",
                         width = 1100, height = 550))
plot(Bar1)

```


Row {data-width=300}
-------------------------------------
### GSOA Results Table
```{r echo = FALSE, results = 'asis'}

# plot results using the datatalbe package
DT::datatable(your_GSOA_results, options = list(bPaginate = F, rownames= T, bFilter= T, bInfo =T, bLengthChange =T))
```

Row {data-width=300}
-------------------------------------

### You ran GSOA with the following parameters 

 **Genomic Data File(s): **  `r params$data_files `         
 **Geneset: **  `r params$genesets`  
 **Machine Learning Algorithm: **  `r params$alg`  
 **Variance Filtered : **  `r params$var`  
 **Low Gene Expression Filtered : **   `r params$lowexpress`  
 **Number of Cross Validations : **  `r params$crossval`  
 **Number of Iterations : ** `r params$iterations`   
 **GSOA Version : ** 1.0 

<br>
<i class="fa fa-calendar"></i>
This report was generated on **`r format(Sys.time(), "%a %b %d %Y %H:%M ")`** 





<!--



# Hallmarks Report {data-icon="fa-life-ring"}

#Row {data-width=300}
-------------------------------------

### Total Number of Gene Sets GSOA Ran
```{r}

#hallmark_GSOA_results= read.table("/app/GSOA_P53_Hallmarks_1000.txt",header=T,sep='\t', quote="", row.names = 1)


#hallmark_GSOA_results= read.table(GSOA,header=T,sep='\t', quote="", row.names = 1)

#hallmark_GSOA_results=GSOA

#total = nrow(hallmark_GSOA_results)

#gauge(total, min = 0, max = nrow(hallmark_GSOA_results) , label = "Gene Sets" )

```
    
### Number of Significant Gene Sets (P-Value)
    
```{r}
#sig_pval = nrow(hallmark_GSOA_results[ which(hallmark_GSOA_results$p.value <= 0.05),])

#gauge(sig_pval, min = 0, max = nrow(hallmark_GSOA_results) , label = "P-value < 0.05", gaugeSectors(
#  success = c(0, total/3), warning = c(total/3+1, total/2), danger = c(total/2+1, sig_pval)))

```

### Number of Significant Gene Sets (FDR)

```{r}
#sig_fdr = nrow(hallmark_GSOA_results[ which(hallmark_GSOA_results$FDR <= 0.05),])
#gauge(sig_fdr, min = 0, max = 100 , label = "FDR < 0.05", gaugeSectors(
#  success = c(0, total/3), warning = c(total/3+1, total/2), danger = c(total/2+1, sig_pval)))
```


Row {data-height=575}
-------------------------------------

### Top 20 Ranked Hallmark Gene Sets

```{r, results='asis', echo=FALSE}

# remove the hallmark from the gene sets so its shorter

#hallmark_GSOA_results_bar= hallmark_GSOA_results
#newrownames= gsub("HALLMARK_", "", row.names(hallmark_GSOA_results))
# hack
#hallmark_GSOA_results_bar[,1]=newrownames

# plot the bar plot
#Bar1 <- gvisBarChart(hallmark_GSOA_results_bar[1:20,], xvar="AUC", yvar=c("Rank", "p.value"), options= list(fontSize=10.5, width = 650, height = 550, title= "Top 20 Gene Sets", legend="right",hAxes="[{title:'Rank', titleTextStyle: {color: 'black', fontSize:14}}]", titleTextStyle="{color:'Black',fontName:'Arial',fontSize:18}",isStacked =  FALSE, chartArea= "{ width: '40%', height: '75%'}" ))
#plot(Bar1)


#Bar <- gvisBarChart(hallmark_GSOA_results_bar[1:20,], "AUC", c("Rank", "p.value"),options=list(title= "Top 20 Gene Sets", legend="bottom",series="[{targetAxisIndex: 1},{targetAxisIndex:0}]",hAxis="{title:'P-value',   titleTextStyle:{color:'blue', fontSize:14}}" , titleTextStyle="{color:'Black',fontName:'Arial',fontSize:18}", chartArea= "{ width: '40%', height: '75%'}" ,
#                        width = 650, height = 550))
#plot(Bar)

```


### Ontology Piechart for Significant Gene Sets (p-value < 0.05)

```{r echo = FALSE, results = 'asis'}
# this file contains the "processes" for the hallmarks 
#hallmarks_processes=read.table("/app/Hallmarks_Geneset_Processes.txt", header=F,sep='\t', quote="", row.names = 1)

#Merge results with proccesses
#merged = merge(hallmarks_processes, hallmark_GSOA_results, by = "row.names")
#row.names(merged)=merged[,1]
#merged=merged[,-1]

#significant_results= merged[which(merged$p.value <= 0.05),]

# Find the freq of each process
#freq = as.data.frame(table(significant_results$V2))

# #remove the processes with a zero
#freq.no0 <- subset(freq, Freq > 0) 

# Plot Pie Chart with Google Viz
#Pie <- gvisPieChart(freq.no0,options= list(title="Hallmark Ontologies", is3D= F, width=500, height=600, fontSize=15, legend ='left', chartArea= "{left: 10, top: 75, width: '100%', height: '100%'}", titleTextStyle="{color:'Black',fontName:'Arial',fontSize:18}"))
#plot(Pie)

```


Row {data-width=300}
-------------------------------------
### GSOA Results Table
```{r echo = FALSE, results = 'asis'}

# plot results using the datatalbe package
#DT::datatable(hallmark_GSOA_results, options = list(bPaginate = F, rownames= T, bFilter= T, bInfo =T, bLengthChange =T))
```


Row {data-width=300}
-------------------------------------

### You ran GSOA with the following parameters 

**Genomic Data File: **  `r params$data1 `         
**Class File: **  `r params$class`  
**Geneset: **  `r params$genesets`  
**Machine Learning Algorithm: **  `r params$alg`  
**Variance Filtered : **  `r params$var`  
**Low Gene Expression Filtered : **   `r params$lowexpress`  
**Number of Cross Validations : **  `r params$crossval`  
**Number of Iterations : ** `r params$iterations`  

<i class="fa fa-calendar"></i>
This report was generated on **`r format(Sys.time(), "%a %b %d %Y %H:%M ")`** 

--> 
